src/mbgl/gl/context.cpp

```diff
@@ -177,8 +177,8 @@ UniqueBuffer Context::createIndexBuffer(const void* data, std::size_t size) {
     BufferID id = 0;
     MBGL_CHECK_ERROR(glGenBuffers(1, &id));
     UniqueBuffer result { std::move(id), { this } };
-    vertexArrayObject = 0;
-    elementBuffer = result;
+    bindVertexArray = 0;
+    globalVertexArrayState.indexBuffer = result;
```

Comment:
My understanding based on https://www.khronos.org/opengl/wiki/Vertex_Specification#Vertex_Buffer_Object is that the vertex buffer binding is _not_ part of VAO state.